### DISCUSSION TITLE
Why psycopg returns string for integer type during SELECT

### DISCUSSION THREAD

**Discussion by soulofmitsui** [psycopg/psycopg2#1439] [Q&A]
Hi,

I noticed in many cases psycopg returns string for integer columns doing SELECT. For example, the count(*) query below returned string (i.e. '1') instead of int. However, the "select oid" query can return int. What's the rule deciding return type? psycopg version is 2.8.6 and python version is 3.8.10.

    cursor.execute("""select count(*) from pg_class where oid = 9927;""")
    results = cursor.fetchall()
    logger.info(results)
    [2022-03-22T22:39:48.357 +0000] | /~er_example.test.py | INFO     | [('1',)]

    cursor.execute("""select oid from pg_class where oid = 9927;""")
    results = cursor.fetchall()
    logger.info(results)
    [2022-03-22T22:41:38.290 +0000] | /~er_example.test.py | INFO     | [(9927,)]

**Comment by dvarrazzo** [comment]
What database is this? I guess not PostgreSQL:

```
In [1]: import psycopg2
In [2]: cnn = psycopg2.connect("")
In [3]: cur = cnn.cursor()
In [4]: cur.execute("select count(*) from pg_class where oid = 9927")
In [5]: cur.fetchone()
Out[5]: (0,)
```

More info about this type can be obtained using:

```
In [6]: cur.execute("select pg_typeof(count(*)) from pg_class where oid = 9927")
In [7]: cur.fetchone()[0]
Out[7]: 'bigint'

In [8]: cur.execute("select pg_typeof(count(*))::int from pg_class where oid = 9927")
In [9]: cur.fetchone()[0]
Out[9]: 20
```

what do you get?

**Reply by soulofmitsui** [reply]
Thanks @dvarrazzo. It turns out it was our internal registered a data processing callback which converted int to strings. Appreciate your help!
